src/openfermion/measurements/_fermion_partitioning.py

```diff
@@ -0,0 +1,389 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import numpy
+from openfermion.measurements import partition_iterator
+
+MAX_LOOPS = 1e6
+
+
+def pair_within(labels):
+    '''
+    Generates a set of len(labels)-1 pairings such that
+    each element in labels is paired with each other element
+    in at least one pairing
```

Comment:
Please be more explicit in the description and provide an example. You should define what you mean by "pairing". "Pairing" sounds like it could mean "pair", which it clearly doesn't.